app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  before_action :set_user_and_post
+
+  def new
+    @comment = Comment.new
+  end
+
+  def create
+    @comment = @post.comments.build(params.require(:comment).permit(:text).merge(user_id: current_user.id))
+    if @comment.save
+      flash[:success] = 'Comment saved successfully'
+      redirect_to user_post_path(@post, @user)
+    else
+      flash.now[:error] = 'Post not saved, try again later'
+      render :new
+    end
+  end
+
+  private
+
+  def set_user_and_post
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:post_id])
+  end
```

Comment:
- [ Optional ] Good Job for A comment is created on behalf of the currently logged in user (current_user) of the app, But Kindly I recommend you to use `the strong parameters`  with the below params method , Please check out this [link](`https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.`) for more info.


```
 def create
    @comment = @post.comments.build(params.merge(user_id: current_user.id))
    if @comment.save
      flash[:success] = 'Comment saved successfully'
      redirect_to user_post_path(@post, @user)
    else
      flash.now[:error] = 'Post not saved, try again later'
      render :new
    end
  end

  private

  def set_user_and_post
    @user = User.find(params[:user_id])
    @post = @user.posts.find(params[:post_id])
  end

def comment_params
    params.require(:comment).permit(:text)
  end

```